pkg/util/log/channels.go

```diff
@@ -42,9 +42,30 @@ func logfDepth(
 			MaybeSendCrashReport(ctx, err)
 		}
 		if ch != channel.OPS {
-			// Tell the OPS channel about this termination.
-			logfDepth(ctx, depth+1, severity.INFO, channel.OPS,
-				"the server is terminating due to a fatal error (see the %s channel for details)", ch)
+			// Tell the OPS channel about this termination, but impose a
+			// timeout on the request so that we can crash in a timely manner
+			// if logging is stalled.
+			exitFunc := func(x exit.Code, _ error) { exit.WithCode(x) }
+			logging.mu.Lock()
+			if logging.mu.exitOverride.f != nil {
+				exitFunc = logging.mu.exitOverride.f
+			}
+			logging.mu.Unlock()
+
+			opsLogFinished := make(chan struct{})
+			go func() {
+				logfDepth(ctx, depth+1, severity.INFO, channel.OPS,
```

Comment:
This changes the stack, need to check that this is OK. It probably is but need to think about it